packages/bot-web-ui/src/components/self-exclusion.jsx

```diff
@@ -0,0 +1,202 @@
+                                                        }}
+                                                        error={errors[field.name] || form_errors[field.name]}
+                                                        hint={localize(
+                                                            'Maximum number of trades your bot will execute for this run.'
+                                                        )}
+                                                    />
+                                                );
+                                            }}
+                                        </Field>
+                                    </Form>
+                                );
+                            }}
+                        </Formik>
+                    </ThemedScrollbars>
+                </div>
+
+                <div className='self-exclusion__footer'>
+                    <Button large text={localize('Cancel')} onClick={ToggleSelfExclusion} secondary />
+                    <Button large text={localize('Apply and run')} onClick={() => onSubmitLimits()} primary />
+                </div>
+            </div>
+        );
+    }
+}
+
+class SelfExclusion extends React.PureComponent {
+    render() {
+        const { is_restricted, ToggleSelfExclusion, is_mobile } = this.props;
+        return (
+            <>
+                {is_mobile ? (
+                    <FadeWrapper is_visible={is_restricted} className='limits__wrapper' keyname='limitis__wrapper'>
+                        <PageOverlay header={localize('Limits')} onClickClose={ToggleSelfExclusion}>
+                            <MobileWrapper>
+                                <Div100vhContainer className='limits__wrapper--is-mobile' height_offset='80px'>
+                                    <SelfExclustionForm {...this.props} />
+                                </Div100vhContainer>
+                            </MobileWrapper>
+                        </PageOverlay>
+                    </FadeWrapper>
+                ) : (
+                    <Modal
+                        is_open={is_restricted}
+                        has_close_icon
+                        width='500px'
+                        toggleModal={ToggleSelfExclusion}
+                        className='self-exclusion__modal'
+                        title={localize('Limits')}
+                    >
+                        <SelfExclustionForm {...this.props} />
+                    </Modal>
+                )}
+            </>
+        );
+    }
+}
+SelfExclusion.propTypes = {
+    checkClientRestriction: PropTypes.func,
+    is_mobile: PropTypes.bool,
+    is_restricted: PropTypes.bool,
+    initial_values: PropTypes.object,
+    max_losses: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    run_limit: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    set_max_losses: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```

Comment:
Shouldn't these values be a number in all cases?

Reply:
No, we discussed about this should be different because when I use it only number have problem by checking onBotRun function (we have same in checking quick strategy)

Reply:
@aarimm, as we discussed before there is a problem when this value is number because the counter on onBotRunningEvent (code bellow) doesn't work fine if I replace "" with 0

if (self_exclusion.run_limit !== '') {
            self_exclusion.run_limit -= 1;
            if (self_exclusion.run_limit < 0) {
                this.onStopButtonClick();
            }
        }